src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java

```diff
@@ -187,12 +211,80 @@ public URL toExternalURL(@NonNull Blob blob, @NonNull HttpMethod httpMethod) thr
 
     @Extension
     public static final class DescriptorImpl extends BlobStoreProviderDescriptor {
+        private static final String BUCKET_REGEXP = "^([a-z]|(\\d(?!\\d{0,2}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})))([a-z\\d]|(\\.(?!(\\.|-)))|(-(?!\\.))){1,61}[a-z\\d\\.]$";
+        private static final Pattern bucketPattern = Pattern.compile(BUCKET_REGEXP);
+
+        public DescriptorImpl() {
+            super();
+        }
+
+        public ListBoxModel doFillRegionItems() {
+            ListBoxModel regions = new ListBoxModel();
+            regions.add("Auto","");
+            for (String s : Region.DEFAULT_S3) {
+                regions.add(s);
+            }
+            return regions;
+        }
+
+        public FormValidation doCheckContainer(@QueryParameter String container){
+            FormValidation ret = FormValidation.ok();
+            if(StringUtils.isBlank(container)){
+                ret = FormValidation.error("The container name cannot be empty");
+            } else if(!bucketPattern.matcher(container).matches()){
+                ret = FormValidation.error("The container name does not match with S3 bucket rules");
+            }
+            return ret;
+        }
+
+        public FormValidation doCheckPrefix(@QueryParameter String prefix){
+            FormValidation ret = FormValidation.ok();
+            if(StringUtils.isBlank(prefix)){
+                ret = FormValidation.error("Prefix should have a value");
+            } else if(!prefix.endsWith("/")){
+                ret = FormValidation.warning("if Prefix point to a folder, it should end with '/' character");
```

Comment:
I think it's better to always add it if not present, better UX

Reply:
but you could use it as refix, I mean, if you set "folder/prefix-", all the stuff will be created as "folder/prefix-WHATEVER" that could be the expected behavior.
If this is not the expected behavior the warning will change to "a '/' will be added to the end"

Reply:
ok, so you are saying that `folder/prefix-` is a valid value and just want to print a warning about it. Fare enough